CASSANDRA-15277 Make it possible to resize concurrent read / write thread pools at runtime

jonmeredith opened the pull request:
See https://issues.apache.org/jira/browse/CASSANDRA-15277

Commit:
Add setCorePoolSize/setMaximumPoolSize to LocalAwareExecs

With implementations for ImmediateExecutor,
JMXEnabledThreadPoolExecutor and SEPExecutor.

Patch by Jon Meredith; Reviewed by ? for CASSANDRA-15277

clohfink commented on src/java/org/apache/cassandra/metrics/CassandraMetricsRegistry.java:
```diff
@@ -186,6 +186,8 @@ else if (metric instanceof Timer)
             mbean = new JmxTimer((Timer) metric, name, TimeUnit.SECONDS, TimeUnit.MICROSECONDS);
         else if (metric instanceof Metered)
             mbean = new JmxMeter((Metered) metric, name, TimeUnit.SECONDS);
+        else if (metric instanceof IntegerSetting)
```
Codahale metrics are typically a read only thing, I think we can certainly go this route but then we have a 1 value in all of Cassandra set this way and I would prefer to see us move to Virtual Tables than to work to refactor all our JMX MBeans. The JMXEnabledThreadPoolExecutor also already has a `setMaximumThreads` so this will be confusing.

nvharikrishna committed:
[CASSANDRA-15227] Removed StageManager.

Commit:
Support resizing the native transport executor

Commit:
Refactor Stage to pass set max pool size consumers

Patch by Jon Meredith; Reviewed by ? for CASSANDRA-15277

Commit:
nodetool: add get/set concurrency commands to nodetool

Commit:
DNM: Setting CircleCI to high res config

Commit:
Introduce ResizableThreadPool interface

Convert JMXEnabledThreadPoolExecutorMBean and SEPExecutorMBean

Commit:
Remove JMXConfigurableThreadPoolExecutor.

JMXConfigurableThreadPoolExecutor is only used by repair,
and JMXEnabledThreadPoolExecutor has been resizable since
CASSANDRA-5044 so there is no longer any need for the specialization
with the ResizableThreadPool now exporting the same API over JMX.

jonmeredith commented:
CASSANDRA-15227 landed, reverting to the original change against the updated Stage and reopening a new PR.
